### PR TITLE
Add Prometheus Reporter 

### PR DESCRIPTION
**Pull request by dasuni-30**
## Purpose
- Add Prometheus Reporter to support publishing to Prometheus for metrics collection.

## Goals
- Support Prometheus for metrics collection to optimize for cloud-based deployments.

## Approach
- Expose the Dropwizard metrics used in Siddhi to Prometheus using DropwizardExports in Prometheus client.
- Use the HTTP server in Prometheus client to expose the metrics using the instrumentation method.

## Release note
Add Prometheus Reporter to support metrics collection.

## Automation tests
 - Unit tests - 85%
 - Integration tests - N/A

## Security checks
- Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
- Ran FindSecurityBugs plugin and verified report? yes
- Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

## Test environment
JDK 1.8, Prometheus

### CODE REVIEW

File: runner/components/io.siddhi.distribution.metrics.core/src/main/java/io/siddhi/distribution/metrics/prometheus/reporter/config/PrometheusReporterConfig.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+        this.serverURL = serverURL;
+    }
+
+    @Override
+    public Optional<PrometheusReporter> build(MetricRegistry metricRegistry, MetricFilter metricFilter)
+            throws ReporterBuildException {
+        if (!isEnabled()) {
+            return Optional.empty();
+        }
+        if (logger.isInfoEnabled()) {
```
Comment (niveathika) [inline runner/components/io.siddhi.distribution.metrics.core/src/main/java/io/siddhi/distribution/metrics/prometheus/reporter/config/PrometheusReporterConfig.java]: Remove isInfoEnabled() check, as it is not needed

File: runner/components/io.siddhi.distribution.metrics.core/src/main/java/io/siddhi/distribution/metrics/prometheus/reporter/config/ReportingConfig.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+ * Configuration for all reporters in Metrics Core.
+ */
+
+public class ReportingConfig {
+
+    private Set<PrometheusReporterConfig> prometheus;
+
+    public ReportingConfig() {
+        this.prometheus = new HashSet<>();
+        this.prometheus.add(new PrometheusReporterConfig());
```
Comment (niveathika) [inline runner/components/io.siddhi.distribution.metrics.core/src/main/java/io/siddhi/distribution/metrics/prometheus/reporter/config/ReportingConfig.java]: This is enabled by default?

File: runner/components/io.siddhi.distribution.metrics.core/src/main/java/io/siddhi/distribution/metrics/prometheus/reporter/impl/PrometheusReporter.java [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+
+        }
+
+    }
+
+    public static PrometheusReporter.Builder forRegistry(MetricRegistry registry, String serverURL) {
+        return new PrometheusReporter.Builder(registry, serverURL);
+    }
+
+    private void initiateServer(String host, int port) throws ConnectionUnavailableException {
```
Comment (niveathika) [inline runner/components/io.siddhi.distribution.metrics.core/src/main/java/io/siddhi/distribution/metrics/prometheus/reporter/impl/PrometheusReporter.java]: This can be moved to the top-level field?

File: runner/components/io.siddhi.distribution.metrics.core/src/main/java/io/siddhi/distribution/metrics/prometheus/reporter/impl/PrometheusReporter.java [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+        private MetricFilter filter;
+        private String serverURL;
+
+        private Builder(MetricRegistry registry, String serverURL) {
+            this.registry = registry;
+            this.serverURL = serverURL;
+            this.filter = MetricFilter.ALL;
+        }
+
+        public Builder setServerURL(String serverURL) {
```
Comment (niveathika) [inline runner/components/io.siddhi.distribution.metrics.core/src/main/java/io/siddhi/distribution/metrics/prometheus/reporter/impl/PrometheusReporter.java]: Is this used?

File: runner/components/io.siddhi.distribution.metrics.core/src/main/resources/META-INF/services/org.wso2.carbon.metrics.core.spi.MetricsExtension [hunk 5]
```diff
@@ -0,0 +1 @@
+io.siddhi.distribution.metrics.prometheus.reporter.PrometheusMetricsExtension
```
Comment (niveathika) [inline runner/components/io.siddhi.distribution.metrics.core/src/main/resources/META-INF/services/org.wso2.carbon.metrics.core.spi.MetricsExtension]: This will be created by default, remove

File: runner/components/io.siddhi.distribution.metrics.core/src/test/java/io/siddhi/distribution/metrics/core/BaseReporterTest.java [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+ */
+public abstract class BaseReporterTest {
+
+    private static final Logger logger = LoggerFactory.getLogger(BaseReporterTest.class);
+    protected static Metrics metrics;
+    protected static MetricManagementService metricManagementService;
+
+    @BeforeSuite
+    protected static void init() throws ConfigurationException {
+        if (logger.isInfoEnabled()) {
```
Comment (niveathika) [inline runner/components/io.siddhi.distribution.metrics.core/src/test/java/io/siddhi/distribution/metrics/core/BaseReporterTest.java]: No need to check for info level log

File: runner/components/io.siddhi.distribution.metrics.core/src/test/java/io/siddhi/distribution/metrics/core/BaseReporterTest.java [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+            logger.info("Creating Metrics");
+        }
+        System.setProperty("metrics.target", "target");
```
Comment (niveathika) [inline runner/components/io.siddhi.distribution.metrics.core/src/test/java/io/siddhi/distribution/metrics/core/BaseReporterTest.java]: Any reason to add this?

File: runner/components/io.siddhi.distribution.metrics.core/src/test/java/io/siddhi/distribution/metrics/core/BaseReporterTest.java [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+        metrics = new Metrics(TestUtils.getConfigProvider("metrics.yaml"));
+        metrics.activate();
+        metricManagementService = metrics.getMetricManagementService();
+    }
+
+    @AfterSuite
+    protected static void destroy() throws Exception {
+        if (logger.isInfoEnabled()) {
```
Comment (niveathika) [inline runner/components/io.siddhi.distribution.metrics.core/src/test/java/io/siddhi/distribution/metrics/core/BaseReporterTest.java]: Same

File: runner/components/io.siddhi.distribution.metrics.core/src/test/java/io/siddhi/distribution/metrics/core/MetricsConfigTest.java [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+ */
+public class MetricsConfigTest {
+
+    private static PrometheusMetricsConfig prometheusMetricsConfig;
+
+    @BeforeClass
+    private void load() throws ConfigurationException {
+        prometheusMetricsConfig = TestUtils.getConfigProvider("metrics-prometheus.yaml")
+                .getConfigurationObject(PrometheusMetricsConfig.class);
+
```
Comment (niveathika) [inline runner/components/io.siddhi.distribution.metrics.core/src/test/java/io/siddhi/distribution/metrics/core/MetricsConfigTest.java]: Same

File: runner/components/io.siddhi.distribution.metrics.core/src/test/java/io/siddhi/distribution/metrics/core/StatisticsTestCase.java [hunk 10]
```diff
@@ -47,6 +46,7 @@
  * Test case for carbon metrics inside siddhi.
  */
 public class StatisticsTestCase {
+
```
Comment (niveathika) [inline runner/components/io.siddhi.distribution.metrics.core/src/test/java/io/siddhi/distribution/metrics/core/StatisticsTestCase.java]: Remove all unnecessary changes

File: runner/components/io.siddhi.distribution.metrics.core/src/test/resources/log4j2.xml [hunk 11]
```diff
@@ -0,0 +1,15 @@
+<?xml version="1.0" encoding="UTF-8"?>
+
+<!--<Configuration>-->
```
Comment (niveathika) [inline runner/components/io.siddhi.distribution.metrics.core/src/test/resources/log4j2.xml]: Remove

**Comment by niveathika** [comment]
@dasuni-30 Do not remove sections on pull request template, Add much information as possible for new features, see https://github.com/siddhi-io/siddhi-store-mongodb/pull/56

File: runner/components/io.siddhi.distribution.metrics.core/pom.xml [hunk 12]
```diff
@@ -69,6 +69,31 @@
...
+        </dependency>
+        <dependency>
+            <groupId>io.prometheus</groupId>
+            <artifactId>simpleclient_dropwizard</artifactId>
+            <version>0.8.0</version>
+        </dependency>
+        <dependency>
+            <groupId>io.dropwizard.metrics</groupId>
+            <artifactId>metrics-core</artifactId>
+            <version>3.1.2</version>
```
[the remark this answers is not in the document]
Comment (dasuni-30) [reply runner/components/io.siddhi.distribution.metrics.core/pom.xml]: The above comment is addressed by b0e138f

File: runner/components/io.siddhi.distribution.metrics.core/src/main/java/io/siddhi/distribution/metrics/prometheus/reporter/PrometheusMetricsExtension.java [hunk 13]
```diff
@@ -0,0 +1,105 @@
...
+    private static final Logger logger = LoggerFactory.getLogger(PrometheusMetricsExtension.class);
+    List<String> reporterNames = new ArrayList<>();
+
+    @Override
+    public void activate(ConfigProvider configProvider, MetricService metricService,
+                         MetricManagementService metricManagementService) {
+        PrometheusMetricsConfig prometheusMetricsConfig;
+        try {
+            prometheusMetricsConfig = configProvider.getConfigurationObject(PrometheusMetricsConfig.class);
+
```
[the remark this answers is not in the document]
Comment (dasuni-30) [reply runner/components/io.siddhi.distribution.metrics.core/src/main/java/io/siddhi/distribution/metrics/prometheus/reporter/PrometheusMetricsExtension.java]: The above comment is addressed by b0e138f

File: runner/components/io.siddhi.distribution.metrics.core/src/main/java/io/siddhi/distribution/metrics/prometheus/reporter/PrometheusMetricsExtension.java [hunk 14]
```diff
@@ -0,0 +1,105 @@
...
+        if (prometheusReporterConfigs != null) {
+            prometheusReporterConfigs.forEach(reporterBuilder -> {
+                        try {
+                            metricManagementService.addReporter(reporterBuilder);
+                        } catch (ReporterBuildException e) {
+                            logger.warn("Prometheus Reporter build failed", e);
+                        }
+                    }
+            );
+            reporterNames = prometheusReporterConfigs.stream().map(prometheusReporterConfig ->
```
[the remark this answers is not in the document]
Comment (dasuni-30) [reply runner/components/io.siddhi.distribution.metrics.core/src/main/java/io/siddhi/distribution/metrics/prometheus/reporter/PrometheusMetricsExtension.java]: The above comment is addressed by b0e138f

File: runner/components/io.siddhi.distribution.metrics.core/src/main/java/io/siddhi/distribution/metrics/prometheus/reporter/PrometheusMetricsExtension.java [hunk 15]
```diff
@@ -0,0 +1,105 @@
...
+
+    @Override
+    public void activate(ConfigProvider configProvider, MetricService metricService,
+                         MetricManagementService metricManagementService) {
+        PrometheusMetricsConfig prometheusMetricsConfig;
+        try {
+            prometheusMetricsConfig = configProvider.getConfigurationObject(PrometheusMetricsConfig.class);
+
+        } catch (ConfigurationException e) {
+            logger.error("Error loading Metrics Configuration. Starting Prometheus Reporter with default parameters."
```
[the remark this answers is not in the document]
Comment (dasuni-30) [reply runner/components/io.siddhi.distribution.metrics.core/src/main/java/io/siddhi/distribution/metrics/prometheus/reporter/PrometheusMetricsExtension.java]: The above comment is addressed by b0e138f

File: runner/components/io.siddhi.distribution.metrics.core/src/main/java/io/siddhi/distribution/metrics/prometheus/reporter/config/PrometheusMetricsConfig.java [hunk 16]
```diff
@@ -0,0 +1,39 @@
...
+ */
+package io.siddhi.distribution.metrics.prometheus.reporter.config;
+
+import org.wso2.carbon.config.annotation.Configuration;
+import org.wso2.carbon.config.annotation.Element;
+
+/**
+ * Configuration for Metrics.
+ */
+@Configuration(namespace = "metrics.prometheus", description = "Carbon Metrics Configuration Parameters")
```
[the remark this answers is not in the document]
Comment (dasuni-30) [reply runner/components/io.siddhi.distribution.metrics.core/src/main/java/io/siddhi/distribution/metrics/prometheus/reporter/config/PrometheusMetricsConfig.java]: The above comment is addressed by b0e138f

File: runner/components/io.siddhi.distribution.metrics.core/src/main/java/io/siddhi/distribution/metrics/prometheus/reporter/config/ReportingConfig.java (hunk 2 above)
Comment (dasuni-30) [reply runner/components/io.siddhi.distribution.metrics.core/src/main/java/io/siddhi/distribution/metrics/prometheus/reporter/config/ReportingConfig.java]: The reporters are not enabled by default.

File: runner/components/io.siddhi.distribution.metrics.core/src/main/java/io/siddhi/distribution/metrics/prometheus/reporter/impl/PrometheusReporter.java (hunk 4 above)
Comment (dasuni-30) [reply runner/components/io.siddhi.distribution.metrics.core/src/main/java/io/siddhi/distribution/metrics/prometheus/reporter/impl/PrometheusReporter.java]: The above comment is addressed by b0e138f

File: pom.xml [hunk 17]
```diff
@@ -1181,7 +1181,7 @@
         <carbon.messaging.version>3.0.3</carbon.messaging.version>
         <carbon.deployment.version>5.2.2</carbon.deployment.version>
         <carbon.datasources.version>1.1.8</carbon.datasources.version>
-        <carbon.metrics.version>2.3.11</carbon.metrics.version>
+        <carbon.metrics.version>2.3.12-SNAPSHOT</carbon.metrics.version>
```
[the remark this answers is not in the document]
Comment (dasuni-30) [reply pom.xml]: The above comment is addressed by the commit 12f0664

File: runner/components/io.siddhi.distribution.metrics.core/pom.xml [hunk 18]
```diff
@@ -133,6 +165,14 @@
                     </execution>
                 </executions>
             </plugin>
+            <plugin>
```
[the remark this answers is not in the document]
Comment (dasuni-30) [reply runner/components/io.siddhi.distribution.metrics.core/pom.xml]: The above comment is addressed by the commit aafbeb1

File: runner/components/io.siddhi.distribution.metrics.core/src/test/resources/log4j2.xml (hunk 11 above)
Comment (dasuni-30) [reply runner/components/io.siddhi.distribution.metrics.core/src/test/resources/log4j2.xml]: The above comment is addressed by the commit 80ebd14.

File: runner/components/io.siddhi.distribution.metrics.core/src/test/java/io/siddhi/distribution/metrics/core/BaseReporterTest.java (hunk 7 above)
Comment (dasuni-30) [reply runner/components/io.siddhi.distribution.metrics.core/src/test/java/io/siddhi/distribution/metrics/core/BaseReporterTest.java]: The above comment is addressed by the commit 84ada78.

File: runner/components/io.siddhi.distribution.metrics.core/src/test/java/io/siddhi/distribution/metrics/core/BaseReporterTest.java (hunk 8 above)
Comment (dasuni-30) [reply runner/components/io.siddhi.distribution.metrics.core/src/test/java/io/siddhi/distribution/metrics/core/BaseReporterTest.java]: The above comment is addressed by the commit 287c655.

File: runner/components/io.siddhi.distribution.metrics.core/src/main/java/io/siddhi/distribution/metrics/prometheus/reporter/config/PrometheusReporterConfig.java (hunk 1 above)
Comment (dasuni-30) [reply runner/components/io.siddhi.distribution.metrics.core/src/main/java/io/siddhi/distribution/metrics/prometheus/reporter/config/PrometheusReporterConfig.java]: The above comment is addressed by the commit 287c655.

File: runner/components/io.siddhi.distribution.metrics.core/src/main/java/io/siddhi/distribution/metrics/prometheus/reporter/impl/PrometheusReporter.java (hunk 3 above)
Comment (dasuni-30) [reply runner/components/io.siddhi.distribution.metrics.core/src/main/java/io/siddhi/distribution/metrics/prometheus/reporter/impl/PrometheusReporter.java]: The above comment is addressed by the commit 88d61ec.

File: runner/components/io.siddhi.distribution.metrics.core/src/test/java/io/siddhi/distribution/metrics/core/BaseReporterTest.java (hunk 6 above)
Comment (dasuni-30) [reply runner/components/io.siddhi.distribution.metrics.core/src/test/java/io/siddhi/distribution/metrics/core/BaseReporterTest.java]: The above comment is addressed by the commit 287c655.

File: runner/components/io.siddhi.distribution.metrics.core/src/test/java/io/siddhi/distribution/metrics/core/StatisticsTestCase.java (hunk 10 above)
Comment (dasuni-30) [reply runner/components/io.siddhi.distribution.metrics.core/src/test/java/io/siddhi/distribution/metrics/core/StatisticsTestCase.java]: The above comment is addressed by the commit 47afa97.

File: runner/components/io.siddhi.distribution.metrics.core/src/main/java/io/siddhi/distribution/metrics/prometheus/reporter/config/PrometheusMetricsConfig.java [hunk 19]
```diff
@@ -0,0 +1,39 @@
...
+public class PrometheusMetricsConfig {
+
+    @Element(description = "Enable Metrics")
```
[the remark this answers is not in the document]
Comment (dasuni-30) [reply runner/components/io.siddhi.distribution.metrics.core/src/main/java/io/siddhi/distribution/metrics/prometheus/reporter/config/PrometheusMetricsConfig.java]: The above comment is addressed by the commit 49936f1.

File: runner/components/io.siddhi.distribution.metrics.core/src/test/java/io/siddhi/distribution/metrics/core/MetricsConfigTest.java (hunk 9 above)
Comment (dasuni-30) [reply runner/components/io.siddhi.distribution.metrics.core/src/test/java/io/siddhi/distribution/metrics/core/MetricsConfigTest.java]: This code segment is necessary to get the configurations from "metrics-prometheus.yaml" file to run the metrics configuration test case.

File: runner/components/io.siddhi.distribution.metrics.core/pom.xml [hunk 20]
```diff
@@ -86,11 +111,18 @@
...
+                            io.siddhi.distribution.metrics.prometheus.*
                         </Export-Package>
                         <Import-Package>
                             *;resolution:=optional,
                         </Import-Package>
+                        <Include-Resource>
+                            META-INF=target/classes/META-INF
+                        </Include-Resource>
+                        <Private-Package>
+                            io.prometheus.client.*
```
[the remark this answers is not in the document]
Comment (dasuni-30) [reply runner/components/io.siddhi.distribution.metrics.core/pom.xml]: The above comment is addressed by the commit 4f9be8b.

[the remark this answers is not in the document]
Comment (niveathika) [reply runner/components/io.siddhi.distribution.metrics.core/pom.xml]: Remove the client from exporting, since you have to import here and not export

File: runner/components/io.siddhi.distribution.metrics.core/src/main/java/io/siddhi/distribution/metrics/prometheus/reporter/PrometheusMetricsExtension.java (hunk 14 above)
[the remark this answers is not in the document]
Comment (niveathika) [reply runner/components/io.siddhi.distribution.metrics.core/src/main/java/io/siddhi/distribution/metrics/prometheus/reporter/PrometheusMetricsExtension.java]: Here, you are adding the reporters again and again. There is no need add only the reported you are adding to the metricsManagement

File: runner/components/io.siddhi.distribution.metrics.core/src/main/java/io/siddhi/distribution/metrics/prometheus/reporter/config/PrometheusMetricsConfig.java (hunk 19 above)
[the remark this answers is not in the document]
Comment (niveathika) [reply runner/components/io.siddhi.distribution.metrics.core/src/main/java/io/siddhi/distribution/metrics/prometheus/reporter/config/PrometheusMetricsConfig.java]: It should be Prometheus reporter config

File: runner/components/io.siddhi.distribution.metrics.core/src/main/java/io/siddhi/distribution/metrics/prometheus/reporter/config/PrometheusReporterConfig.java (hunk 1 above)
Comment (niveathika) [reply runner/components/io.siddhi.distribution.metrics.core/src/main/java/io/siddhi/distribution/metrics/prometheus/reporter/config/PrometheusReporterConfig.java]: Format the code

File: runner/components/io.siddhi.distribution.metrics.core/src/main/java/io/siddhi/distribution/metrics/prometheus/reporter/impl/PrometheusReporter.java (hunk 4 above)
Comment (niveathika) [reply runner/components/io.siddhi.distribution.metrics.core/src/main/java/io/siddhi/distribution/metrics/prometheus/reporter/impl/PrometheusReporter.java]: No, its seems to be an unused method. any reason to add it?

File: runner/components/io.siddhi.distribution.metrics.core/src/main/java/io/siddhi/distribution/metrics/prometheus/reporter/impl/PrometheusReporter.java (hunk 3 above)
Comment (niveathika) [reply runner/components/io.siddhi.distribution.metrics.core/src/main/java/io/siddhi/distribution/metrics/prometheus/reporter/impl/PrometheusReporter.java]: Catch MalformedUrlException too, else it will not be handled

File: runner/components/io.siddhi.distribution.metrics.core/src/main/java/io/siddhi/distribution/metrics/prometheus/reporter/PrometheusMetricsExtension.java [hunk 21]
```diff
@@ -0,0 +1,105 @@
...
+            prometheusMetricsConfig = new PrometheusMetricsConfig();
+        }
+        Set<PrometheusReporterConfig> prometheusReporterConfigs = prometheusMetricsConfig.getReporting()
+                .getPrometheus();
+        if (prometheusReporterConfigs != null) {
+            prometheusReporterConfigs.forEach(reporterBuilder -> {
+                        try {
+                            metricManagementService.addReporter(reporterBuilder);
+                        } catch (ReporterBuildException e) {
+                            logger.warn("Prometheus Reporter build failed", e);
```
[the remark this answers is not in the document]
Comment (dasuni-30) [reply runner/components/io.siddhi.distribution.metrics.core/src/main/java/io/siddhi/distribution/metrics/prometheus/reporter/PrometheusMetricsExtension.java]: The above comment is addressed by the commit 2fceb3e

File: runner/components/io.siddhi.distribution.metrics.core/src/main/java/io/siddhi/distribution/metrics/prometheus/reporter/config/PrometheusMetricsConfig.java (hunk 19 above)
[the remark this answers is not in the document]
Comment (dasuni-30) [reply runner/components/io.siddhi.distribution.metrics.core/src/main/java/io/siddhi/distribution/metrics/prometheus/reporter/config/PrometheusMetricsConfig.java]: The above comment is addressed by the commit 7faa6b3.

File: runner/components/io.siddhi.distribution.metrics.core/src/main/java/io/siddhi/distribution/metrics/prometheus/reporter/impl/PrometheusReporter.java (hunk 3 above)
Comment (dasuni-30) [reply runner/components/io.siddhi.distribution.metrics.core/src/main/java/io/siddhi/distribution/metrics/prometheus/reporter/impl/PrometheusReporter.java]: MalformedUrlException is also caught in the above catch statement.

File: runner/components/io.siddhi.distribution.metrics.core/src/main/java/io/siddhi/distribution/metrics/prometheus/reporter/config/PrometheusReporterConfig.java (hunk 1 above)
Comment (dasuni-30) [reply runner/components/io.siddhi.distribution.metrics.core/src/main/java/io/siddhi/distribution/metrics/prometheus/reporter/config/PrometheusReporterConfig.java]: The above comment is addressed by the commit 4e68f2a.

File: runner/components/io.siddhi.distribution.metrics.core/src/main/java/io/siddhi/distribution/metrics/prometheus/reporter/impl/PrometheusReporter.java (hunk 4 above)
Comment (dasuni-30) [reply runner/components/io.siddhi.distribution.metrics.core/src/main/java/io/siddhi/distribution/metrics/prometheus/reporter/impl/PrometheusReporter.java]: The above comment is addressed by the commit 9a0bfe0.

File: runner/components/io.siddhi.distribution.metrics.core/src/main/resources/META-INF/services/org.wso2.carbon.metrics.core.spi.MetricsExtension (hunk 5 above)
Comment (niveathika) [reply runner/components/io.siddhi.distribution.metrics.core/src/main/resources/META-INF/services/org.wso2.carbon.metrics.core.spi.MetricsExtension]: Removed in eb52aca

File: runner/components/io.siddhi.distribution.metrics.core/pom.xml (hunk 20 above)
[the remark this answers is not in the document]
Comment (niveathika) [reply runner/components/io.siddhi.distribution.metrics.core/pom.xml]: As discussed offline, let's pack the Prometheus clients here, and export them

[the remark this answers is not in the document]
Comment (dasuni-30) [reply runner/components/io.siddhi.distribution.metrics.core/pom.xml]: Private-Package the clients in the metrics core. The above change is addressed by the commit 36f6ae2.
